src/pages/index.js

```diff
@@ -2,46 +2,84 @@
 import FormValidator from "../components/FormValidator.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
+import PopupConfirm from "../components/PopupConfirm.js";
+import { validatorConfig, cardsListSelector, apiData } from "../utils/constants.js";
 import "../pages/index.css";
-import { initialCards, validatorConfig, cardsListSection } from "../utils/constants.js";
 
 //select elements
 const profile = document.querySelector(".profile");
 const profileEditButton = profile.querySelector(".profile__edit-button");
 const profileAddButton = profile.querySelector(".profile__add-button");
-const profileName = profile.querySelector(".profile__title");
-const profileJob = profile.querySelector(".profile__description");
+const profileNameEl = profile.querySelector(".profile__title");
+const profileAboutEl = profile.querySelector(".profile__description");
+const profileAvatar = profile.querySelector(".profile__image");
+const profileAvatarEdit = profile.querySelector(".profile__image-edit");
 const profileModalForm = document.forms["profile-form"];
 const profileModalNameInput = profileModalForm.querySelector(".profile-modal__name-input");
 const profileModalDescInput = profileModalForm.querySelector(".profile-modal__desc-input");
 const addModalForm = document.forms["add-form"];
+const avatarEditForm = document.forms["avatar-edit-form"];
 
 //instantiate classes
-const userInfo = new UserInfo(profileName, profileJob);
+const userInfo = new UserInfo({
+  name: profileNameEl,
+  about: profileAboutEl,
+  avatar: profileAvatar,
+  apiHeaders: apiData.headers,
+  apiUser: apiData.currentUser,
+});
+userInfo.getUserInfo();
+
+const initialCardsApi = new Api({
+  url: apiData.cards,
+  headers: apiData.headers,
+});
+initialCardsApi.handleFetch().then((res) => {
+  cardsListSection.renderItems(res, true);
+});
+
+const cardsListSection = new Section((item, firstRender) => {
+  cardsListSection.addItem(createCard(item), firstRender);
+}, cardsListSelector);
+
 const profileFormValidator = new FormValidator(validatorConfig, profileModalForm);
 const addFormValidator = new FormValidator(validatorConfig, addModalForm);
+const avatarEditFormValidator = new FormValidator(validatorConfig, avatarEditForm);
 profileFormValidator.enableValidation();
 addFormValidator.enableValidation();
+avatarEditFormValidator.enableValidation();
 
 const profilePopup = new PopupWithForm(
   "#profile-modal",
   (inputFieldValues, evt) => {
     evt.preventDefault();
-    userInfo.setUserInfo(inputFieldValues);
+    userInfo.setUserInfo(inputFieldValues, profilePopup);
   },
   profileFormValidator
 );
 profilePopup.setEventListeners();
 
+const avatarEditPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  (inputValue, evt) => {
+    evt.preventDefault();
+    profileAvatar.src = inputValue.Link;
+    // evt.submitter.textContent = "Saving...";
+    updateAvatar(profileAvatar.src);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```